temporal101/Exercise1.hs

```diff
@@ -0,0 +1,87 @@
+module Main where
+
+import Control.Monad (forever)
+import Control.Monad.IO.Class (liftIO)
+import Control.Monad.Logger (defaultOutput)
+import Control.Monad.Trans.Reader (runReaderT)
+import Data.Aeson (FromJSON, ToJSON)
+import Data.Text (Text)
+import Data.UUID qualified as UUID
+import Data.UUID.V4 qualified as UUID.V4
+import DiscoverInstances (discoverInstances)
+import GHC.Generics (Generic)
+import RequireCallStack (RequireCallStack, provideCallStack)
+import System.IO (stdout)
+import Temporal.Activity (Activity)
+import Temporal.Client (mkWorkflowClientConfig, workflowClient)
+import Temporal.Client qualified as Client
+import Temporal.Core.Client (connectClient, defaultClientConfig)
+import Temporal.Duration (seconds)
+import Temporal.Payload (JSON (JSON))
+import Temporal.Runtime (TelemetryOptions (..), initializeRuntime)
+import Temporal.TH (WorkflowFn, ActivityFn)
+import Temporal.TH qualified
+import Temporal.Worker (Worker, WorkerConfig, startWorker)
+import Temporal.Worker qualified as Worker
+import Temporal.Workflow (Workflow, WorkflowId (..))
+import Temporal.Workflow qualified as Workflow
+import UnliftIO.Concurrent (threadDelay)
+import UnliftIO.Exception (bracket)
+
+-- | The official "Hello, Workflow" example takes a bare string as input.
+-- Here we reinforce the pattern of creating a record type for inputs so
+-- that it can evolve freely.
+data SayHelloInput = SayHelloInput
+  { name :: Text
+  }
+  deriving stock (Generic, Show)
+  deriving anyclass (FromJSON, ToJSON)
+
+-- | We don't execute an activity in this workflow, we execute code
+-- directly (why?).
+sayHelloWorkflow :: SayHelloInput -> Workflow Text
+sayHelloWorkflow input = provideCallStack do
+  pure $ "Hello, " <> input.name
+
+Temporal.TH.registerWorkflow 'sayHelloWorkflow
+Temporal.TH.bringRegisteredTemporalFunctionsIntoScope
+
+taskQueue :: Workflow.TaskQueue
+taskQueue = "hello-world"
+
+namespace :: Workflow.Namespace
+namespace = "default"
+
+workerConfig :: WorkerConfig ()
+workerConfig = provideCallStack $ Worker.configure environment definitions settings
+  where
+    environment = ()
+    definitions :: RequireCallStack => Worker.Definitions ()
+    definitions = Temporal.TH.discoverDefinitions @() $$(discoverInstances) $$(discoverInstances)
+    settings = do
+      Worker.setNamespace namespace
+      Worker.setTaskQueue taskQueue
+      Worker.setLogger (defaultOutput stdout)
+
+main :: IO ()
+main = bracket setup teardown $ \_worker -> do
+  -- the worker's running in its own thread, we just have to keep this
+  -- one alive (one one-second threadDelay at a time) for as long as we
+  -- want to execute workflows on it
+  forever $ threadDelay 1_000_000
+  pure ()
+  where
+    setup = do
+      runtime <- initializeRuntime NoTelemetry
+      -- coreClient is needed to construct a worker, even though we're
+      -- not constructing a client
+      coreClient <- connectClient runtime defaultClientConfig
```

Comment:
I bumped the version of `hs-temporal-sdk` we're building against, so you'll need to add `runStdoutLoggingT` here to discharge the `MonadLogger` constraint that was added to `connectClient`